include/kernel_version.h

```diff
@@ -30,6 +30,62 @@ extern "C" {
 #define SYS_KERNEL_VER_MINOR(ver) (((ver) >> 16) & 0xFF)
 #define SYS_KERNEL_VER_PATCHLEVEL(ver) (((ver) >> 8) & 0xFF)
 
+struct version_header {
```

Comment:
There are no dependencies between the existing code in kernel_version.h/version.c and the new 'version_header' feature.

To cleanly separate the two features this new feature should be placed in it's own header and source pair.

Doing this will allow us to:

- reduce the number of #ifdef's
- Reduce the noise in kernel_version.h for users that do not use CONFIG_VERSION_HEADER
- Make it clear how version_header and sys_kernel_version interact (they don't)

Reply:
Ok.